### PR TITLE
Password room issue bugfix #49

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java [hunk 1]
```diff
@@ -18,16 +19,28 @@
...
  * @author PTS34A
  */
 public final class JoinSessionMessageHandler extends MessageHandler<JoinSessionMessage> {
 
     @Override
     protected void handle(Connection connection, JoinSessionMessage message) throws Exception {
         Status status = message.getStatus();
-        if (status != Status.SUCCESS) {
-            return;
+        System.out.println(status.name());
```
Comment (BeeHiveJava) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: Gooby pls.

**Review by BeeHiveJava — CHANGES_REQUESTED** [review]

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java [hunk 2]
```diff
@@ -85,4 +98,13 @@ protected JoinSessionMessage decode(Connection connection, ByteBuf buf) throws E
         return new JoinSessionMessage(status, roomName, capacity, settings);
     }
 
+    private void failedJoiningOfSession(String title, String message) {
+        Platform.runLater(() -> {
```
Comment (TehMessiah) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: MAJOR Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1602)

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java (hunk 1 above)
Comment (TehMessiah) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS106)

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java [hunk 3]
```diff
@@ -18,16 +19,28 @@
...
  */
 public final class JoinSessionMessageHandler extends MessageHandler<JoinSessionMessage> {
 
     @Override
     protected void handle(Connection connection, JoinSessionMessage message) throws Exception {
         Status status = message.getStatus();
-        if (status != Status.SUCCESS) {
-            return;
+        System.out.println(status.name());
+        switch (status) {
```
Comment (TehMessiah) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: MAJOR Add a default case to this switch. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java (hunk 2 above)
Comment (BeeHiveJava) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: He's right too. ^

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java [hunk 4]
```diff
@@ -41,6 +40,8 @@ protected void handle(Connection connection, JoinSessionMessage message) throws
             case USERNAME_EXISTS:
                 failedJoiningOfSession("Username exists", "There is already a player with the same name in the Room, please change your username.");
                 return;
+            default:
```
Comment (BeeHiveJava) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java]: Exception toevoegen (UnsupportedOperationException).

**Review by BeeHiveJava — CHANGES_REQUESTED** [review]

File: Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java [hunk 5]
```diff
@@ -18,16 +19,32 @@
...
 /**
  * Handles JoinSession messages, after the currentPlayer has joined a session.
- * 
+ *
  * @author PTS34A
  */
 public final class JoinSessionMessageHandler extends MessageHandler<JoinSessionMessage> {
 
     @Override
     protected void handle(Connection connection, JoinSessionMessage message) throws Exception {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/handler/JoinSessionMessageHandler.java:28]: MAJOR The Cyclomatic Complexity of this method "handle" is 11 which is greater than 10 authorized. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by ghost** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Review by BeeHiveJava — APPROVED** [review]

**Pull request merged**